lib/fileHandler/index.js

```diff
@@ -0,0 +1,117 @@
+var knox   = require('knox');
+var when   = require('when');
+var guid   = require('mout/random/guid');
+var util   = require('util');
+var needle = require('needle');
+var mmm    = require('mmmagic');
+
+
+module.exports = function (options) {
+
+  if (!_.isObject(options.aws)) {
+    return;
+  }
+
+  /*
+  * Upload a file to S3.
+  */
+  var upload = function (params) {
+    return when.promise(function (resolve, reject) {
+
+      var client = knox.createClient({
+        key: options.aws.key,
+        secret: options.aws.secret,
+        bucket: 'workflow-file-uploads',
+        region: 'us-west-2'
+      });
+
+
+      // Generate a random file name
+      var fileName = guid();
+      var buffer;
+      if (!util.isBuffer(params.contents)) {
+        return when.reject(new Error('Please pass the `contents` as a Node.js Buffer.'))
+      } else {
+        buffer = params.contents;
+      }
+
+      // Detect the file type
+      var magic = new mmm.Magic(mmm.MAGIC_MIME_TYPE);
+      magic.detect(buffer, function (err, mimeType) {
+        if (err) {
+          reject(err);
+        } else {
+
+          // TODO: should we pass the content-type to S3?
+          var headers = {
+            'Content-Type': mimeType
+          };
+
+          // Stick the buffer in S4
+          client.putBuffer(buffer, fileName, headers, function (err, res) {
+            if (err) {
+              reject(err);
+            } else if (res.statusCode !== 200) {
+              reject(new Error(res.statusCode));
+            } else {
+              var fifteenMinutes = new Date(Date.now() + 15 * 60000);
```

Comment:
I reckon just set it to 24 hours for now.. i think its an ok amount of time, and we can set the bucket TTL to that as well